h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -897,38 +894,20 @@ private final double lmax(double[] grad) {
 
 
     private void doCleanup() {
-      try {
-        if(_parms._lambda_search && _parms._is_cv_model)
-          Scope.untrack(removeLater(_dinfo.getWeightsVec()._key));
-        if (_parms._HGLM) {
-          Key[] vecKeys = _toRemove;
-          for (int index=0; index < vecKeys.length; index++) {
-            Vec tempVec = DKV.getGet(vecKeys[index]);
-            tempVec.remove();
-          }
+      if(_parms._lambda_search && _parms._is_cv_model)
+        Scope.untrack(removeLater(_dinfo.getWeightsVec()._key));
+      if (_parms._HGLM) {
+        Key[] vecKeys = _toRemove;
+        for (int index=0; index < vecKeys.length; index++) {
+          Vec tempVec = DKV.getGet(vecKeys[index]);
+          tempVec.remove();
         }
-        if(_model!=null)
-          _model.unlock(_job);
-      } catch(Throwable t){
```

Comment:
this seem reckless, I think we want to know if something breaks during cleanup

Reply:
I think you have a valid point, however, I think this code needs more error handling if this catch is removed, eg.:

Can _dinfo be null? Can the temp vec already be removed from DKV (line 903 possible NPE).

I propose this:

Add the error handling
in onCompletion - invoke doCleanup the regular way
in onExceptionalCompletion - we are in an undefined state - any clean-up attempt is by definition best effort only - I would keep the try-catch - and produce a warning in the log

Reply:
you are correct, such a change is out of scope for this PR, hence I have reverted most of it and at least added some logging

Reply:
it was `Throwable` really? at least it should be limited to `Exception`